controllers/remote/cluster.go

```diff
@@ -26,12 +27,16 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	defaultClientTimeout = 10 * time.Second
+)
+
 // ClusterClientGetter returns a new remote client.
-type ClusterClientGetter func(ctx context.Context, c client.Client, cluster client.ObjectKey) (client.Client, error)
+type ClusterClientGetter func(ctx context.Context, sourceName string, c client.Client, cluster client.ObjectKey) (client.Client, error)
 
 // NewClusterClient returns a Client for interacting with a remote Cluster using the given scheme for encoding and decoding objects.
-func NewClusterClient(ctx context.Context, c client.Client, cluster client.ObjectKey) (client.Client, error) {
-	restConfig, err := RESTConfig(ctx, c, cluster)
+func NewClusterClient(ctx context.Context, sourceName string, c client.Client, cluster client.ObjectKey) (client.Client, error) {
+	restConfig, err := RESTConfig(ctx, sourceName, c, cluster)
```

Comment:
These changes make the PR a breaking change ⚠️

Reply:
so are you saying we should not do that? or document it? I didn't how else to identity the base component calling

Reply:
I was waiting if see if reviewers think this approach is good before updating the PR description to say its breaking

@detiber is that more in the direction you had in mind?

Reply:
Mostly document it

Reply:
This also means this PR should not be backported to 0.3 (not sure if there were plans to)

Reply:
I don't think there are any plans to back port this one